File: webapp-net/Ugc/Models/UgcComment.cs
using Sdl.Web.Common.Models;
using Sdl.Web.Modules.Ugc.Data;
using System.Collections.Generic;

namespace Sdl.Web.Modules.Ugc.Models
{
    public class UgcComment : EntityModel
    {
        public Comment CommentData { get; set; }

        public List<UgcComment> Comments { get; set; }
    }
}

File: webapp-java/dxa-module-ugc/src/main/java/com/sdl/dxa/modules/ugc/model/entity/UgcComment.java
package com.sdl.dxa.modules.ugc.model.entity;

import com.sdl.dxa.modules.ugc.data.Comment;
import com.sdl.webapp.common.api.model.entity.AbstractEntityModel;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.List;

/**
 * <p>EntityModel for a single Ugc comment</p>
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class UgcComment extends AbstractEntityModel {

    /**
     * Comment Data
     **/
    private Comment commentData;

    /**
     * List of related comments
     **/
    private List<UgcComment> comments;

}
